benchmarks/NN/functional_c#/Program.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.Linq;
+using System.Collections.Immutable;
+
+
+namespace functional_c_
+{
+    using Layer = ImmutableArray<Neuron>;
+    struct Neuron
+    {
+        public readonly ImmutableArray<double> weights;
+        public readonly double output;
+        public readonly double bias;
+        public readonly double delta;
+        public Neuron(ImmutableArray<double> w, double o, double b, double d){
+            weights = w;
+            output = o;
+            bias = b;
+            delta = d;
+        }
+    }
+
+    class Program
+    {
+        static Random rand = new Random(2);
+
+        static readonly ImmutableArray<ImmutableArray<double>> rawData = System.IO.File.ReadAllLines("benchmarks/NN/wheat-seeds.csv").Select(l => l.Split(',').Select(n => double.Parse(n)).ToImmutableArray()).ToImmutableArray();
+        static readonly ImmutableArray<ImmutableArray<double>> wheatData = rawData.Select(row => row.SetItem(row.Length - 1, row[^1] - 1)).ToImmutableArray();
+
+        static void Main(string[] args)
+        {
+            var nInput = wheatData[0].Length;
+            var nHidden = 5;
+            var nOutput = wheatData.Select(row => row[row.Length - 1]).Distinct().Count();
+            var learningRate = 0.3;
+            var epochs = 500;
+
+            var dataset = normaliseDataset(wheatData, datasetMinMax(wheatData))
+                        .OrderBy(_ => rand.Next())
+                        .ToImmutableArray();
+            
+            var splitIndex = dataset.Length / 10;
+            var testData = dataset.Take(splitIndex).ToImmutableArray();
+            var trainData = dataset.TakeLast(dataset.Length - splitIndex).ToImmutableArray();
+
+            var init = initialiseNetwork(nInput, nHidden, nOutput);
+
+            var trainedNetwork = trainNetwork(init, trainData, learningRate, epochs, nOutput);
+ 
+            var results = testData.Select(row => (prediction: predict(trainedNetwork, row), actual: (int)row[^1])).ToImmutableArray();
+            var correctPredictions = results.Where(res => res.prediction == res.actual).Count();
+            var accuracy = correctPredictions / (double)results.Length * 100.0;
+            
+            System.Console.WriteLine(accuracy);
+        }
+
+        private static ImmutableArray<Layer> backwardsPropagateError(ImmutableArray<Layer> network, ImmutableArray<double> expected)
+        {
+            var firstLayer = network.First();
+            var remainingLayers = network.TakeLast(network.Length - 1);

```

Comment:
Jeg tror takelast er lineær, ville det her ikke være bedre:
```suggestion
            var remainingLayers = network[1 ..];

```